engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -0,0 +1,115 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedObjectTypingResult, TypingResult}
+import cats.instances.list._
+
+import scala.collection.JavaConverters._
+import pl.touk.nussknacker.engine.util.Implicits._
+import pl.touk.nussknacker.engine.util.validated.ValidatedSyntax
+
+object aggregates {
+
+  private val syntax = ValidatedSyntax[String]
+  import syntax._
+
+  object MaxAggregator extends ReducingAggregator {
+
+    override type Element = Number
+
+    override def zero: Number = Double.MinValue
+
+    override def add: (Number, Number) => Number = (n1, n2) => Math.max(n1.doubleValue(), n2.doubleValue())
+
+    override def zeroType: TypingResult = Typed[Number]
+
+  }
+
+  object MinAggregator extends ReducingAggregator {
+
+    override type Element = Number
+
+    override def zero: Number = Double.MaxValue
+
+    override def add: (Number, Number) => Number = (n1, n2) => Math.min(n1.doubleValue(), n2.doubleValue())
+
+    override def zeroType: TypingResult = Typed[Number]
+  }
+
+  object ListAggregator extends Aggregator {
+
+    override type Aggregate = List[AnyRef]
+
+    override type Element = AnyRef
+
+    override def zero: Aggregate = List()
+
+    override def add: (Element, Aggregate) => Aggregate = _ :: _
+
+    override def merge: (Aggregate, Aggregate) => Aggregate = _ ++ _
+
+    override def result: Aggregate => Any = _.asJava
+
+    override def computeOutputType(input: TypingResult): Validated[String, TypingResult]
+      = Valid(TypedClass(classOf[java.util.List[_]], List(input)))
+
+  }
+
+  class MapAggregator(fields: java.util.Map[String, Aggregator]) extends Aggregator {
```

Comment:
Can you give some example how it works? You can have max on some field and min on some other? How it handle null in values? Will omit aggregations?

Reply:
I'll add some samples and tests

Reply:
comments/test added, I added TODO for null handling, I think some more general consideration would be needed - e.g. can we implement #input.complexField.?field1? :D